rclcpp/include/rclcpp/client.hpp

```diff
@@ -44,15 +46,23 @@ class ClientBase
 public:
   RCLCPP_MAKE_SHARED_DEFINITIONS(ClientBase);
 
-  ClientBase(rmw_client_t * client_handle, const std::string & service_name)
-  : client_handle_(client_handle), service_name_(service_name)
+  ClientBase(
+    std::shared_ptr<rmw_node_t> node_handle,
+    rmw_client_t * client_handle,
+    const std::string & service_name)
+  : node_handle_(node_handle), client_handle_(client_handle), service_name_(service_name)
   {}
 
   ~ClientBase()
   {
-    if (client_handle_ != nullptr) {
-      rmw_destroy_client(client_handle_);
-      client_handle_ = nullptr;
+    if (client_handle_) {
+      if (rmw_destroy_client(client_handle_) == RMW_RET_ERROR) {
+        // *INDENT-OFF*
+        std::cerr << "Error in destruction of rmw client handle: "
+                  << (rmw_get_error_string() ? rmw_get_error_string() : "")
```

Comment:
Maybe it should be something like `"error string not set"`. In reality the else case shouldn't happen, this is just defensive so that we don't trying to send a null pointer to `std::cerr`.

I also think we should encapsulate this logic in a function, maybe `rmw_get_error_string_safe` or something like that, where it would guarantee that cstring is returned (`""` or `"error string not set"`) and a null pointer is not. Currently it can return null if it is not set.